web/src/message_list.ts

```diff
@@ -17,6 +20,24 @@ import {user_settings} from "./user_settings";
 
 export class MessageList {
     static id_counter = 0;
+
+    id: number;
+    data: MessageListData;
+    // The MessageListView object that is responsible for
+    // maintaining this message feed's HTML representation in the
+    // DOM.
+    view: MessageListView;
+    // If this message list is for the combined feed view.
+    is_combined_feed_view: boolean;
+    // Keeps track of whether the user has done a UI interaction,
+    // such as "Mark as unread", that should disable marking
+    // messages as read until prevent_reading is called again.
+    //
+    // Distinct from filter.can_mark_messages_read(), which is a
+    // property of the type of narrow, regardless of actions by
+    // the user. Possibly this can be unified in some nice way.
+    reading_prevented: boolean;
+    last_message_historical?: boolean;
     // A MessageList is the main interface for a message feed that is
     // rendered in the DOM. Code outside the message feed rendering
     // internals will directly call this module in order to manipulate
```

Comment:
This comment should be first, since it's describing what the class is for.

Reply:
Moved